app/models/user.rb

```diff
@@ -320,6 +320,10 @@ def human_authentication_provider
     authentication_provider&.display_name
   end
 
+  def provided_by_oidc?
+    authentication_provider.is_a?(OpenIDConnect::Provider)
```

Comment:
Future-proofing hints:

1. We'll soon have the possibility of more than one provider associated to the same user (logging in through multiple was always possible, but until recently we only stored the most recent provider)
2. The fact that an `authentication_provider` is present, doesn't even mean that the user used it to log-in

The more I think about this, the more I believe that these problems are out of scope for your changes here. So take this as a mere FYI ^^

Reply:
Yeah, this will need to be updated once we have the multiple auth providers in. =/